src/views/List.jsx

```diff
@@ -1,23 +1,47 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
-	const listItems = data.map((data) => {
-		return <ListItem key={data.id} name={data.name} />;
-	});
+	const [currentItems, setCurrentItems] = useState('');
+	const [list, setList] = useState([]);
+	const [input, setInput] = useState('');
+
+	function handleInputChange(e) {
+		setCurrentItems(e.target.value);
+		setInput(e.target.value);
+	}
+
+	useEffect(() => {
+		const filteredItems = data.filter((item) =>
+			item.name.includes(currentItems),
+		);
+		const newListItems = filteredItems.map((item) => (
+			<ListItem key={item.id} name={item.name} />
+		));
+		setList(newListItems);
+	}, [data, currentItems]);
+
+	function clearSearch() {
+		setCurrentItems('');
+		setInput('');
+	}
 
 	return (
 		<>
+			<form action="">
+				<label htmlFor="searchItems">Search Items: </label>
+				<input
+					onChange={handleInputChange}
+					id="searchItems"
+					value={input}
+					type="text"
+				/>
+			</form>
+			<button onClick={clearSearch}>X</button>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
-			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
-				{listItems}
-			</ul>
+			<ul>{list}</ul>
```

Comment:
Instead of having JSX in your state, I would suggest mapping and returning JSX based on your state data
So something like this: 
```
			<ul>
				{filteredItems.map((item) => {
					return <ListItem key={item.id} name={item.name} />;
				})}
			</ul>
```